src-docs/src/components/guide_section/_utils.js

```diff
@@ -80,7 +79,7 @@ export const renderJsSourceCode = (code) => {
     // import                        - import + whitespace
     // ([^]+?)                       - capture any characters (including newlines)
     //  from ('[A-Za-z0-9 -_.@/]*';) - ` from 'someLibrary';` - alphanumeric and certain special characters only
-    /(\/\/.+\n)?import ([^]+?) from ('[A-Za-z0-9 -_.@/]*';)/g,
+    /(\/\/.+\n)?import ((?!React)[^]+?) from ('[A-Za-z0-9 -_.@/]*';)/g,
```

Comment:
While this works, I'm not a fan because it's not actually targeting the issue we want to solve - which is to say, it's finding an `import` string _within a code snippet_ that isn't actually an import. So if the import snippet in question changed to `import foo from 'bar'`, we'd still have the exact same issue.

I would strongly rather fix the [negative lookbehind](https://javascript.info/regexp-lookahead-lookbehind#lookbehind) to find **any** backtick character before **any** `import` statement and not `React` specifically. You should be able to accomplish this like so:

```suggestion
    /(?<!(`[\s\S]*))(\/\/.+\n)?import ([^]+?) from ('[A-Za-z0-9 -_.@/]*';)/g,
```

Since regexes are incredibly hard to read, we should also document this new addition in the code above line 78:

```js
    // (?<!(`[\s\S]*))               - negative lookbehind ensuring that import statements aren't part of a template literal
    // (\/\/.+\n)?                   - optional preceding comments that must be above specific imports, e.g. // @ts-ignore

Reply:
There should be no scenario where a real import comes after a template literal backtick - our current eslint rules ensure that all legitimate `import`s are at the top of the file, so this is a fairly safe assertion to make.